content/blog/2020/2020-04-02-covid-19-basic.md

```diff
@@ -0,0 +1,205 @@
+
+- Transpose transposes the table. Now we have countries in columns and dates in rows. Open the second Data Table widget to check.
+- In the Preprocess widget, we drag Normalize features to the right pane and select normalization to [0, 1]. It will linearly scale the data to fit into [0, 1], which effectively divides each column with the population.
+- The second Transpose puts the data into its initial form -- countries in rows, days in columns.
+- In the last Select Columns, we remove the Population column. See the third Data Table widget to check. Check whether the values make sense.
+- In the Line Plot, we'll look at the curves for cases per million people. Well, not really -- it's the number of cases per million divided by a million. In other words, the proportion of the infected population, which is kind of the same thing as far as we're concerned.
+
+I told you it would be a hack. But it's cool, isn't it?
+
+## Normalized curves, via the normal approach
+
+When Orange can't do something, there's always the Python Script widget.
+
+{{% figure src="/blog_img/2020/2020-04-02-workflow-pyscript.png" %}}
+
+The start of the sequence is the same as before: we merge the two data sources and remove countries with unknown or zero population.
+
+In Select Columns, we move all variables except daily data from features to metas.
+
+In Python Script, we type the following program and run it:
+
+```python
+import Orange
+
+normalized = in_data.X / in_data[:, "Total Population (millions) 2015"].metas
+
+out_data = Orange.data.Table.from_numpy(
+    in_data.domain,
+    normalized, in_data.Y, in_data.metas)
+```
+
+With `in_data[:, "Total Population (millions) 2015"]` we take the table corresponding to the column with the population. Since population is now a meta attribute, we take `.metas`. We divide `in_data.X` with this column. Finally, we construct a new table named `out_data`, with the same domain (that is, the same variables), the normalized data, and the original data for target variable(s) and metas.
+
+<!!! LINE PLOT SCREENSHOT !!!>
+
+Observe the line plot. There are some countries with 3000 confirmed cases per million?! Which? Check yourself!
+
+Would you like to see a plot with a logarithmic axis? And Line Plot doesn't support it? (Yet?) The beauty of using the Python script widget is that we can transform the data any way we want. Change the line that computes the normalized data to
+
+```python
+normalized = np.log10(in_data.X / in_data[:, "Total Population (millions) 2015"].metas)
+```
+
+And we have a logarithmic axis (just with wrong labels).
```

Comment:
with wrong labels --> with the wrong labels

Reply:
Here I wouldn't put "the". These are "some" wrong labels.